.github/workflows/ci.yaml

```diff
@@ -53,257 +53,35 @@ permissions:
-              echo "${dest}" >> $GITHUB_STEP_SUMMARY
-            else
-              manifests=$(
-                docker manifest inspect ${source} |\
-                jq -r '.manifests[] | select(.platform.os != "unknown") | .digest' |\
-                xargs -I{} echo ${source_repo}@{} |\
-                tr '\n' ' '
-              )
-
-              ## registry/org/repo:tag -> repo-tag
-              # dest_tag=$(echo ${source} | cut -d: -f1 | cut -d/ -f3)-$(echo ${source} | cut -d: -f2)
-              ## registry/org/repo:tag -> tag
-              dest_tag=$(echo ${source} | cut -d: -f2)
-
-              for manifest in ${manifests}; do
-                os=$(docker manifest inspect -v $manifest | jq -r '.Descriptor.platform.os')
-                arch=$(docker manifest inspect -v $manifest | jq -r '.Descriptor.platform.architecture')
-                # single_arch_tag="ghcr.io/nvidia/jax-toolbox-retrofit:${{ github.run_id }}-${dest_tag}-${os}-${arch}"
-                single_arch_tag="${DOCKER_REPO}:${dest_tag}-${os}-${arch}"
-                skopeo copy --format v2s2 docker://$manifest docker://${single_arch_tag}
-                echo "${single_arch_tag}" >> $GITHUB_STEP_SUMMARY
-              done
-            fi
-          done
-
-  test-distribution:
-    needs: metadata
-    uses: ./.github/workflows/_test_distribution.yaml
-    secrets: inherit
-
-  test-jax:
-    needs: build-jax
-    uses: ./.github/workflows/_test_jax.yaml
-    with:
-      JAX_IMAGE: ${{ needs.build-jax.outputs.DOCKER_TAGS }}
-    secrets: inherit
-
-  test-te:
-    needs: build-te
-    uses: ./.github/workflows/_test_te.yaml
-    with:
-      JAX_TE_IMAGE: ${{ needs.build-te.outputs.DOCKER_TAGS }}
-    secrets: inherit
-  test-t5x:
-    needs: build-t5x
-    uses: ./.github/workflows/_test_t5x.yaml
-    with:
-      T5X_IMAGE: ${{ needs.build-t5x.outputs.DOCKER_TAGS }}
-    secrets: inherit
-
-  test-pax:
-    needs: build-pax
-    uses: ./.github/workflows/_test_pax.yaml
+ 
+  arm64:
+    uses: ./.github/workflows/_ci.yaml
     with:
-      PAX_IMAGE: ${{ needs.build-pax.outputs.DOCKER_TAGS }}
+      ARCHITECTURE: arm64
+      CUDA_IMAGE: ${{ inputs.CUDA_IMAGE || 'latest' }}
+      SRC_JAX: ${{ inputs.SRC_JAX || 'https://github.com/google/jax.git#main' }}
+      SRC_XLA: ${{ inputs.SRC_XLA || 'https://github.com/openxla/xla.git#main'}}
+      SRC_TE: ${{ inputs.SRC_TE || 'https://github.com/NVIDIA/TransformerEngine.git#main'}}
+      SRC_T5X: ${{ inputs.SRC_T5X || 'https://github.com/google-research/t5x.git#main'}}
+      SRC_PAXML: ${{ inputs.SRC_PAXML || 'https://github.com/google/paxml.git#main'}}
+      SRC_PRAXIS: ${{ inputs.SRC_PRAXIS || 'https://github.com/google/praxis.git#main'}}
     secrets: inherit
 
-  test-vit:
-    needs: build-rosetta-t5x
-    uses: ./.github/workflows/_test_vit.yaml
-    with:
-      ROSETTA_T5X_IMAGE: ${{ needs.build-rosetta-t5x.outputs.DOCKER_TAGS }}
-    secrets: inherit 
- 
   finalize:
```

Comment:
Are you planning on adding a "merge" step here to merge the arm64 and amd64 into a multiarch manifest? Or is the expectation for the concrete workflows `"Nightly ..."`  to provide a "ARCHITECTURE" workflow input?

I'm testing #354 and realizing that if a concrete workflow uses a multiarch image, it will fail using the image from the presubmit, since it's a single arch tag.

Reply:
Right now we are doing one job per multi-arch image (JAX/PAX/T5X) and it's too verbose. We could add a single job at the end of the presubmit to properly group all single-arch images. It can take advantage of the sitrep jsons to locate the single-arch images.

Reply:
Would it be OK to do it as a follow-up PR?

Reply:
Sure. Let's not block this PR